File: CS/Sample/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using Sample.Models;

namespace Sample.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();

        [HttpGet]
        public ActionResult Index() {
            return View(list.GetPersons());
        }

        [HttpPost]
        public ActionResult Index(string selectedIDsHF) {
            //Get all selected keys from hidden input
            string _selectedIDs = selectedIDsHF;

            return View(list.GetPersons());
        }

        public ActionResult GridViewEditingPartial() {
            //Get all selected keys from e.customArgs on GridView callback
            string _selectedIDs = Request.Params["selectedIDs"];
            ViewData["_selectedIDs"] = _selectedIDs;

            return PartialView(list.GetPersons());
        }

    }
}

File: VB/Sample/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports Sample.Models

Namespace Sample.Controllers
	Public Class HomeController
		Inherits Controller
		Private list As New PersonsList()

		<HttpGet> _
		Public Function Index() As ActionResult
			Return View(list.GetPersons())
		End Function

		<HttpPost> _
		Public Function Index(ByVal selectedIDsHF As String) As ActionResult
			'Get all selected keys from hidden input
			Dim _selectedIDs As String = selectedIDsHF

			Return View(list.GetPersons())
		End Function

		Public Function GridViewEditingPartial() As ActionResult
			'Get all selected keys from e.customArgs on GridView callback
			Dim _selectedIDs As String = Request.Params("selectedIDs")
			ViewData("_selectedIDs") = _selectedIDs

			Return PartialView(list.GetPersons())
		End Function

	End Class
End Namespace